### PR TITLE
FreeApplicative#hoist is same as compile

### PR DESCRIPTION
**Pull request by adelbertc**

### CODE REVIEW

**Comment by fthomas** [comment]
`compile` requires an `Applicative[G]` but `hoist` does not. If they are the same why don't we remove the one which is more restricted?

**Comment by adelbertc** [comment]
Good catch - `Applicative` constraint was not necessary :-) The 2 are the same

File: free/src/main/scala/cats/free/FreeApplicative.scala [hunk 1]
```diff
@@ -45,7 +38,8 @@ sealed abstract class FreeApplicative[F[_], A] extends Product with Serializable
   final def fold(implicit F: Applicative[F]): F[A] =
     foldMap(NaturalTransformation.id[F])
 
-  final def compile[G[_]](f: F ~> G)(implicit G: Applicative[G]): FA[G, A] =
+  /** Interpreter this FreeApplicative algebra into another using FreeApplicative. */
```
Comment (ceedubs) [inline free/src/main/scala/cats/free/FreeApplicative.scala]: Minor typo here. I think you probably meant "Interpret this..."

**Comment by ceedubs** [comment]
:+1:

**Comment by non** [comment]
:+1:

**Pull request merged**